src/main/kotlin/com/lykke/me/subscriber/rabbitmq/MeRabbitMqProtoEventListener.kt

```diff
@@ -0,0 +1,117 @@
+package com.lykke.me.subscriber.rabbitmq
+
+import com.lykke.me.subscriber.config.RabbitMqConfig
+import com.lykke.me.subscriber.incoming.events.proto.MeProtoEvent
+import com.lykke.me.subscriber.incoming.events.proto.deserialization.MeProtoEventDeserializer
+import com.lykke.utils.logging.ThrottlingLogger
+import com.lykke.utils.notification.AbstractListener
+import com.lykke.utils.rabbit.Connector
+import com.lykke.utils.rabbit.ConsumerFactory
+import com.lykke.utils.rabbit.RabbitMqSubscriber
+import com.rabbitmq.client.Channel
+import com.rabbitmq.client.ConnectionFactory
+import com.rabbitmq.client.Consumer
+import java.util.concurrent.BlockingQueue
+import kotlin.concurrent.thread
+import com.lykke.utils.rabbit.RabbitMqConfig as UtilsRabbitMqConfig
+
+class MeRabbitMqProtoEventListener(private val configs: Set<RabbitMqConfig>,
+                                   private val queue: BlockingQueue<RmqMessageWrapper>) : AbstractListener<MeProtoEvent<*>>() {
+
+    companion object {
+        private val LOGGER = ThrottlingLogger.getLogger(MeRabbitMqProtoEventListener::class.java.name)
+    }
+
+    private var started = false
+    private val deserializerByRoutingKey = HashMap<String, MeProtoEventDeserializer<*>>()
+
+    @Synchronized
+    fun start() {
+        if (started) {
+            return
+        }
+        initDeserializers()
+        startRabbitMqSubscribers()
+        startMessagesProcessing()
+        started = true
+    }
+
+    private fun initDeserializers() {
+        deserializerByRoutingKey.putAll(configs.map { it.routingKey }
+                .groupBy { it }
+                .mapValues { MeProtoEventDeserializer.createDeserializer(it.value.single()) })
+    }
+
+    private fun startRabbitMqSubscribers() {
+        configs.forEach(::startRabbitMqSubscriber)
+    }
+
+    private fun startRabbitMqSubscriber(config: RabbitMqConfig) {
+        val tmpFactory = ConnectionFactory()
+        tmpFactory.setUri(config.uri)
```

Comment:
as I understand currently a user of the lib have only able to use default username/password.